### PR TITLE
add handle_ping/1 pool callback

### PR DESCRIPTION
**Pull request by oliveigah**
closes https://github.com/dashbitco/nimble_pool/issues/46

Let me know what you think about this approach @josevalim!

### CODE REVIEW

File: lib/nimble_pool.ex [hunk 1]
```diff
@@ -565,7 +592,9 @@ defmodule NimblePool do
         state: pool_state,
         lazy: lazy,
         worker_idle_timeout: worker_idle_timeout,
-        max_idle_pings: max_idle_pings
+        max_idle_pings: max_idle_pings,
+        pool_idle_timeout: pool_idle_timeout,
+        last_activity_ts: System.monotonic_time(:millisecond)
```
Comment (josevalim) [inline lib/nimble_pool.ex:597]: Yes, this is exactly my concern. If we have to track this, this is now state necessary on every single nimble pool... but not all of them may need it. It is better if you track this on Finch side.

Comment (oliveigah) [reply lib/nimble_pool.ex:597]: Although I think the implementation overhead is very low, I agree that not all clients may need it.

So it’s all good with me. Feel free to close this PR and the issue, and I’ll move forward with the implementation on the Finch side. Thanks for the feedback!

Comment (josevalim) [reply lib/nimble_pool.ex:597]: Thank you for trying it out! And, if the Finch side for some reason is not enough, we already have what is necessary to implement it here. :)

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request closed without merging**